src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -77,12 +79,21 @@ public BusinessesService(List<Listing> allBusinesses) {
             .apiKey(KEY)
             .build();
     TextSearchRequest request = PlacesApi.textSearchQuery(context, product);
-    
+    int radius = 1500;
+
     try {
       PlacesSearchResponse response = request.location(latLng)
-              .radius(TEXT_SEARCH_RADIUS)
+              .radius(radius)
               .await();
-      
+
+      while (response.results.length < MAX_NUMBER_OF_RESULTS_PER_REQUEST && 
+             radius < MAX_ALLOWED_TEXT_SEARCH_RADIUS) {
+        // If the list of businesses are not enough, the radius is increased
+        // in hopes to get more results for that product search      
+        radius *= RADIUS_MULTIPLIER;
+        response = request.location(latLng).radius(radius).await();
```

Comment:
nit: let's try to make the formatted consistent with the response on line 85 and line 94 (either way)

Reply:
Since the line is less then 80 characters it might be better to leave it as it is